src/backend/distributed/master/master_repair_shards.c

```diff
@@ -87,31 +98,36 @@ master_copy_shard_placement(PG_FUNCTION_ARGS)
 	int32 targetNodePort = PG_GETARG_INT32(4);
 	bool doRepair = PG_GETARG_BOOL(5);
 	Oid shardReplicationModeOid = PG_GETARG_OID(6);
-	char shardReplicationMode = LookupShardTransferMode(shardReplicationModeOid);
 
 	char *sourceNodeName = text_to_cstring(sourceNodeNameText);
 	char *targetNodeName = text_to_cstring(targetNodeNameText);
 
-	if (!doRepair)
-	{
-		ereport(ERROR, (errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
-						errmsg("master_copy_shard_placement() "
-							   "with do not repair functionality "
-							   "is only supported on Citus Enterprise")));
-	}
-	else if (shardReplicationMode == TRANSFER_MODE_FORCE_LOGICAL)
+	CheckCitusVersion(ERROR);
+	EnsureCoordinator();
+
+	char shardReplicationMode = LookupShardTransferMode(shardReplicationModeOid);
+	if (shardReplicationMode == TRANSFER_MODE_FORCE_LOGICAL)
 	{
 		ereport(ERROR, (errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
-						errmsg("using logical replication with repair functionality "
-							   "is currently not supported")));
+						errmsg("using logical replication in "
+							   "master_copy_shard_placement() requires Citus "
+							   "Enterprise")));
 	}
 
-	EnsureCoordinator();
-	CheckCitusVersion(ERROR);
+	ShardInterval *shardInterval = LoadShardInterval(shardId);
+	ErrorIfTableCannotBeReplicated(shardInterval->relationId);
 
-	/* RepairShardPlacement function repairs only given shard */
-	RepairShardPlacement(shardId, sourceNodeName, sourceNodePort, targetNodeName,
-						 targetNodePort);
+	if (doRepair)
+	{
+		RepairShardPlacement(shardId, sourceNodeName, sourceNodePort, targetNodeName,
```

Comment:
At some point it would be nice if repair would behave like copy except it can skip all active placements.